src/py/mat3ra/made/tools/calculate.py

```diff
@@ -57,56 +62,67 @@ def calculate_surface_energy(slab: Atoms, bulk: Atoms, calculator: Calculator):
         calculator (ase.calculators.calculator.Calculator): The calculator to use for the energy calculation.
 
     Returns:
         float: The adhesion energy of the interface.
     """
     energy_substrate_slab = calculate_total_energy(substrate_slab, calculator)
-    energy_layer_slab = calculate_total_energy(layer_slab, calculator)
+    energy_film_slab = calculate_total_energy(film_slab, calculator)
     energy_interface = calculate_total_energy(interface, calculator)
     area = get_surface_area(interface)
-    return (energy_substrate_slab + energy_layer_slab - energy_interface) / area
+    return (energy_substrate_slab + energy_film_slab - energy_interface) / area
 
 
-@decorator_convert_material_args_kwargs_to_atoms
 def calculate_interfacial_energy(
-    interface: Atoms,
-    substrate_slab: Atoms,
-    substrate_bulk: Atoms,
-    layer_slab: Atoms,
-    layer_bulk: Atoms,
-    calculator: Calculator,
+    interface: Material,
+    substrate_slab: Optional[Material] = None,
+    substrate_bulk: Optional[Material] = None,
+    film_slab: Optional[Material] = None,
+    film_bulk: Optional[Material] = None,
+    calculator: Calculator = EMT(),
 ):
     """
     Calculate the interfacial energy.
-    The interfacial energy is the sum of the surface energies of the substrate and layer minus the adhesion energy.
+    The interfacial energy is the sum of the surface energies of the substrate and film minus the adhesion energy.
     According to Dupré's formula
 
     Args:
-        interface (ase.Atoms): The interface Atoms object to calculate the interfacial energy of.
-        substrate_slab (ase.Atoms): The substrate slab Atoms object to calculate the interfacial energy of.
-        substrate_bulk (ase.Atoms): The substrate bulk Atoms object to calculate the interfacial energy of.
-        layer_slab (ase.Atoms): The layer slab Atoms object to calculate the interfacial energy of.
-        layer_bulk (ase.Atoms): The layer bulk Atoms object to calculate the interfacial energy of.
-        calculator (ase.calculators.calculator.Calculator): The calculator to use for the energy calculation.
+        interface (Material): The interface Material object to calculate the interfacial energy of.
+        substrate_slab (Material): The substrate slab Material object to calculate the interfacial energy of.
+        substrate_bulk (Material): The substrate bulk Material object to calculate the interfacial energy of.
+        film_slab (Material): The film slab Material object to calculate the interfacial energy of.
+        film_bulk (Material): The film bulk Material object to calculate the interfacial energy of.
+        calculator (ase.calculators.calculator.Calculator): The calculator to use for the energy calculation
 
     Returns:
         float: The interfacial energy of the interface.
     """
-
+    substrate_slab = filter_by_label(interface, 0) if substrate_slab is None else substrate_slab
```

Comment:
`filter_by_label(interface, 0)` should be isolated as `get_slab(type="substrate")` in `interface/utils`

Reply:
And should use the map object that we have elsewhere for 0, 1 mapping

Reply:
yes. Maybe later this can be a method on `class Interface(Material)`